torch/distributed/tensor/parallel/style.py

```diff
@@ -55,6 +56,21 @@ def __init__(self) -> None:
         super().__init__(make_input_replicate_1d, make_output_tensor)
 
 
+class PairwiseSequenceParallel(ParallelStyle):
+    """
+    PairwiseParallel concatenate colwise and rowwise styles as a fixed
+    pair like what Megatron-LM(https://arxiv.org/abs/1909.08053) is doing.
+    We assume both input and output needs to a replicate DTensor.
```

Comment:
both input and output are replicate DTensors? or both input and output needs to become a replicate DTensor?

Reply:
Opps.. I forgot to update the comments. To answer your question, both input and output should be a Sharded DTensor, but in reality we don't do redistribute for input DTensor (From Sharded to Replicated) in matmul.. Let me think how we can better approach this.